cli/tests/upgrade.bats

```diff
@@ -157,11 +162,121 @@ EOF
+  skip "FIXME: upgrades package but reports that no packages were upgraded"
+  export FLOX_FEATURES_USE_CATALOG=true
+
+  "$FLOX_BIN" init
+  _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_old_hello.json" "$FLOX_BIN" install hello
+
+  old_hello_response_rev=$(jq -r '.[0].[0].pages.[0].page' "$CATALOG_RESPONSES/install_old_hello.json")
+  old_hello_locked_rev=$(jq -r '.packages.[0].rev_count' "$LOCK_PATH")
+  assert_equal "$old_hello_locked_rev" "$old_hello_response_rev"
+
+  export _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_hello.json"
+  run "$FLOX_BIN" upgrade toplevel
+  assert_output --partial "Upgraded 'hello'"
+  hello_response_rev=$(jq -r '.[0].[0].pages.[0].page' "$CATALOG_RESPONSES/install_hello.json")
+  hello_locked_rev=$(jq -r '.packages.[0].rev_count' "$LOCK_PATH")
+  assert_equal "$hello_locked_rev" "$hello_response_rev"
+  
+  assert_not_equal "$old_hello_locked_rev" "$hello_locked_rev"
+}
+
+@test "catalog: upgrade by iid" {
+  skip "FIXME: upgrades package but reports that no packages were upgraded"
+  export FLOX_FEATURES_USE_CATALOG=true
+
+  "$FLOX_BIN" init
+  _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_old_hello.json" "$FLOX_BIN" install hello
+
+  old_hello_response_rev=$(jq -r '.[0].[0].pages.[0].page' "$CATALOG_RESPONSES/install_old_hello.json")
+  old_hello_locked_rev=$(jq -r '.packages.[0].rev_count' "$LOCK_PATH")
+  assert_equal "$old_hello_locked_rev" "$old_hello_response_rev"
+
+  export _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_hello.json"
+  run "$FLOX_BIN" upgrade hello
+  assert_output --partial "Upgraded 'hello'"
+  hello_response_rev=$(jq -r '.[0].[0].pages.[0].page' "$CATALOG_RESPONSES/install_hello.json")
+  hello_locked_rev=$(jq -r '.packages.[0].rev_count' "$LOCK_PATH")
+  assert_equal "$hello_locked_rev" "$hello_response_rev"
+  
+  assert_not_equal "$old_hello_locked_rev" "$hello_locked_rev"
+}
+
+@test "catalog: upgrade errors on iid in group with other packages" {
+  skip "FIXME: upgrades package but reports that no packages were upgraded"
+  export FLOX_FEATURES_USE_CATALOG=true
+
+  "$FLOX_BIN" init
+  _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_curl_hello.json" "$FLOX_BIN" install curl hello
+
+  export _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_hello.json"
+  run "$FLOX_BIN" upgrade hello
+  assert_output --partial "package in the group 'toplevel' with multiple packages"
+}
+
+@test "catalog: check confirmation when all packages are up to date" {
+  # This will pass right now, but that's because _all_ of these tests get
+  # the "no packages need to be upgraded" message. Unskip once we get a real
+  # signal from these tests
+  skip "FIXME: upgrades package but reports that no packages were upgraded"
+  export FLOX_FEATURES_USE_CATALOG=true
+
+  "$FLOX_BIN" init
+  _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_curl_hello.json" "$FLOX_BIN" install curl hello
 
+  export _FLOX_USE_CATALOG_MOCK="$CATALOG_RESPONSES/install_curl_hello.json"
```

Comment:
question nonblocking: in the ticket you said most of these could be made unit tests - did you just decide it's not worth the effort?

Reply:
Yeah, I’d like to revisit in the future when we have time to make better fixtures for manifests and lock files in the Rust side of the test suite